adapter/PreciceInterface.c

```diff
@@ -188,18 +187,18 @@ void Precice_ReadCouplingData( SimulationData * sim )
 				setFaceHeatTransferCoefficients( interfaces[i]->faceCenterData, interfaces[i]->numElements, interfaces[i]->xloadIndices, sim->xload );
 				break;
             case FORCES:
-                //Read and set forces as point loads
+                // Read and set forces as concentrated loads (Neumann BC)
                 precicec_readBlockVectorData( interfaces[i]->forcesDataID, interfaces[i]->numNodes, interfaces[i]->preciceNodeIDs, interfaces[i]->nodeVectorData );
-                setNodeForces( interfaces[i]->nodeVectorData, interfaces[i]->numNodes, interfaces[i]->xforcIndices, sim->xforc );
+                setNodeForces( interfaces[i]->preciceNodeIDs, interfaces[i]->nodeVectorData, interfaces[i]->numNodes, interfaces[i]->xforcIndices, sim->xforc);
                 break;
             case DISPLACEMENTS:
-                printf( "Displacements cannot be used as read data\n" );
-                fflush( stdout );
-                exit( EXIT_FAILURE );
-                break;
-	    case DISPLACEMENTDELTAS:
+				// Read and set displacements as single point constraints (Dirichlet BC)
+				precicec_readBlockVectorData( interfaces[i]->displacementsDataID, interfaces[i]->numNodes, interfaces[i]->preciceNodeIDs, interfaces[i]->nodeVectorData );
+				setNodeDisplacements( interfaces[i]->nodeVectorData, interfaces[i]->numNodes, interfaces[i]->xbounIndices, sim->xboun );
+				break;
+	    	case DISPLACEMENTDELTAS:
                 printf( "DisplacementDeltas cannot be used as read data\n" );
-                fflush( stdout );
+				fflush( stdout );
```

Comment:
White spaces seem to have been mixed up a little. Please correct this (at several positions in the code).